### DISCUSSION TITLE
Possible to run .air games on Android x64?

### DISCUSSION THREAD

**Discussion by Roadhog360** [airsdk/Adobe-Runtime-Support#704] [Q&A]
Title. I have been trying to get an Adobe AIR game working. For Windows I'd just install the runtime and install the .air file to my computer. For Android however I've been hitting nothing but dead ends. Is there any way to get .air games to install and/or run on my phone?

**Comment by ajwfrost** [comment]
On Android the application developer needs to create an APK file; .air files won't work on Android because of the things that need to happen in order to create an "application" from the mobile OS perspective. Same on iOS.. for mobile platforms, there need to be custom packages created for each platform. Potentially, if you have the .air file and have installed it, then you may have everything you need to create the APK yourself via the AIR SDK..

In case you also wanted to know about Ubuntu: for Linux, there was a shared runtime from Adobe (albeit a very old version of AIR) but they took this down at the end of last year. We've got AIR working on Linux but haven't got the 'shared runtime' capabilities where it would do the necessary installation steps. Once the AIR SDK is available for Linux it should be possible to use this to create a 'native' installation of a game based on an .air file though.

**Reply by itlancer** [reply]
@ajwfrost, when AIR version for Linux plans to be available?

**Reply by Roadhog360** [reply]
> On Android the application developer needs to create an APK file; .air files won't work on Android because of the things that need to happen in order to create an "application" from the mobile OS perspective. Same on iOS.. for mobile platforms, there need to be custom packages created for each platform. Potentially, if you have the .air file and have installed it, then you may have everything you need to create the APK yourself via the AIR SDK..
> 
> In case you also wanted to know about Ubuntu: for Linux, there was a shared runtime from Adobe (albeit a very old version of AIR) but they took this down at the end of last year. We've got AIR working on Linux but haven't got the 'shared runtime' capabilities where it would do the necessary installation steps. Once the AIR SDK is available for Linux it should be possible to use this to create a 'native' installation of a game based on an .air file though.

Thanks for the reply! I actually figured out my issue for Ubuntu and that's why I edited it out. I simply used Wine on the installer for the runtime. Then I opened the .air with the runtime installer. Super simple stuff.

Is there any tutorial I could use to turn an installed air application into an apk?

**Reply by ajwfrost** [reply]
If you copy the installed application out of "Program Files (x86)" and put it somewhere that you can then tinker.. try:
1. Delete the "Adobe AIR" folder if that exists
2. Copy the "application.xml" file from META-INF\AIR to the root folder, and delete the "META-INF" folder
3. Delete the 'mimetype' file if that exists
4. Edit the "application.xml" file and change the "supportedProfiles" value to "mobileDevice". You may also want to add some permissions here, perhaps, it depends what this application is actually doing...
5. Try packaging it, e.g.:
`adt -package -target apk -storetype pkcs12 -keystore xxx.p12 MyApplication.apk application.xml .`
(make sure you have the trailing '.' = "include all files from the current folder")

This kind of worked when I tried it on an app here, at least it installed.. but I don't really get the view that I was expecting.. I believe that's because the app is just assuming that there are certain feature available to desktop applications that are not available on Android, for example NativeMenu.
You'll need to create a self-signed certificate if you haven't got one already, e.g.
`adt -certificate -cn MyName 2048-RSA xxx.p12 password`

**Reply by Roadhog360** [reply]
How do I create the certificate? Just replace the placeholders with my name and a chosen number? Is that it?

**Reply by ajwfrost** [reply]
Should just be a case of changing "MyName" to your own name or company name ("Common Name" is the field) and then 'xxx' with whatever filename you want to output. Keep the 2048 number, that's the 2048-bit certificate (you can still generate 1024-bit ones if you want to but they're not really used now..) - and change "password" to something more complex :-)

There are some other options e.g. "-ou org-unit", "-o org-name", "-c country", "-validityPeriod numYears", but they can be left as defaults...
